includes/admin/class-wc-admin-setup-wizard.php

```diff
@@ -386,11 +391,16 @@ class="location-input wc-enhanced-select dropdown"
 			>
 				<option value=""><?php esc_html_e( 'Choose a currency&hellip;', 'woocommerce' ); ?></option>
 				<?php foreach ( get_woocommerce_currencies() as $code => $name ) : ?>
-					<option value="<?php echo esc_attr( $code ); ?>" <?php selected( $currency, $code ); ?>>
+					<option value="<?php echo esc_attr( $code ); ?>">
 						<?php printf( esc_html__( '%1$s (%2$s)', 'woocommerce' ), $name, get_woocommerce_currency_symbol( $code ) ); ?>
 					</option>
 				<?php endforeach; ?>
 			</select>
+			<input
+				id="currency_by_country"
+				type="hidden"
+				data-map="<?php esc_attr_e( json_encode( $currency_by_country ) ); ?>"
+			/>
```

Comment:
Is there a reason not to use WP's `wp_localize_script` to pass the `$currency_by_country` array to JS?

Reply:
I'm not sure — how would we use it here? I think it performs l10n on the server and is intended for static display strings?

However, I do now realize there definitely is reason to use `<script type="text/javascript">` to define a global variable instead of this `<input type="hidden">` which gets submitted (but any reason not to?)

Reply:
Using `wp_localize_script()` is a common way in WP to pass PHP variables as global js variables. You can see an example of it in use in WCS here:

https://github.com/Automattic/woocommerce-services/blob/master/classes/class-wc-connect-nux.php#L432

In addition to passing localized strings, we're also passing variables like `redirect_url`. In this case, we ended up with a global variable called `wcs_nux_notice`, with some properties including `redirect_url`. Then, it can be used in JS like so:

https://github.com/Automattic/woocommerce-services/blob/master/client/banner.js#L80

Reply:
+1 to `wp_localize_script()` instead of a hidden input.

Reply:
Sounds good — I tried but it didn't work:
```php
wp_localize_script( 'wc-setup', 'wc_setup_currencies', $currency_by_country );
```

Putting it in the `setup_wizard` function after registering the script does work, so is there some reason it wouldn't work from within a function called with `call_user_func`?